gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/DefaultTokenStateService.java

```diff
@@ -264,4 +280,29 @@ protected String getTokenDisplayText(final String token) {
     return String.format(Locale.ROOT, "%s...%s", token.substring(0, 10), token.substring(token.length() - 3));
   }
 
+  /**
+   * Method that deletes expired tokens based on the token timestamp.
+   */
+  protected void evictExpiredTokens() {
+    try {
+      for (final String token : getTokens()) {
+        if (isExpired(token)) {
+          log.evictToken(getTokenDisplayText(token));
+          removeToken(token);
+        }
+      }
+    } catch(final Exception e) {
```

Comment:
Would it not be better to catch the error for each eviction attempt, and continue trying the others when one fails, rather than failing the whole attempt because of a single error?

Reply:
Sure, that can be done.